### PR TITLE
pipeline for orb publish

### PR DESCRIPTION
**Pull request by svenwb**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+      - orb-tools/validate:
+          orb-path: src/@orb.yml
 
 workflows:
-  lint-validate:
+  publish:
     jobs:
       - orb-tools/lint
+      - validate
+      - orb-tools/publish-dev:
```
Comment (SilentGert) [inline .circleci/config.yml]: Is there a difference between publish and publish-dev?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+          checkout: true
+          publish-token-variable: CIRCLECI_TOKEN
+          requires:
+            - orb-tools/lint
+            - validate
+          filters:
+            branches:
+              ignore:
+                - master
+      - orb-tools/increment:
```
Comment (SilentGert) [inline .circleci/config.yml]: Do you have any dependency between increment and publish? Without looking up what it does I assume it updates the version number of the orb?

File: src/@orb.yml [hunk 3]
```diff
@@ -80,13 +87,10 @@ jobs:
           attach-workspace: true
           repo: <<parameters.ecr_repo>>
           tag: <<parameters.tags>>
+          dockerfile: <<parameters.dockerfile>>
```
Comment (karlderkaefer) [inline src/@orb.yml]: where is this parameter used?

File: src/@orb.yml [hunk 4]
```diff
@@ -80,13 +87,10 @@ jobs:
...
       - run:
           name: "install kubectl"
```
Comment (karlderkaefer) [inline src/@orb.yml]: stage name does not fit action. maybe do separate step `install-tools` and `rollout`

File: .circleci/config.yml (hunk 1 above)
Comment (svenwb) [reply .circleci/config.yml]: `publish-dev` tags with dev:alpha and overrides this tag. Publish requires a semver which cannot be overwritten later

File: src/@orb.yml (hunk 3 above)
Comment (svenwb) [reply src/@orb.yml]: this parameter gets passed to the aws-ecr/build-and-push-image command. It is helpful when your Dockerfile is not named `Dockerfile` or not the default location

File: src/@orb.yml (hunk 4 above)
Comment (svenwb) [reply src/@orb.yml]: fixed

**Pull request merged**